tedge/src/cli/connect/command.rs

```diff
@@ -284,7 +312,7 @@ async fn check_connection_azure(port: u16) -> Result<(), ConnectError> {
     match fut.await {
         Ok(Ok(true)) => {
             println!("Received expected response message, connection check is successful.");
-            Ok(())
+            Ok(DeviceStatus::New)
```

Comment:
Why always new here?

Reply:
I assumed the response code 200 means new. Is there a different response code for already existing? Can we detect the difference between a new device and an already existing device from the response codes in Azure?

Reply:
200 means success, means the device exists. I feel the check_connection_az works properly just to check that the connection to the device works or not. May be not the case with c8y if I understand correctly. So, its better to return here DeviceStatus::alreadyExists/exists than New.